server/service/service_osquery.go

```diff
@@ -446,15 +447,79 @@ func (svc *Service) detailQueriesForHost(ctx context.Context, host *fleet.Host)
 }
 
 func (svc *Service) shouldUpdate(lastUpdated time.Time, interval time.Duration, hostID uint) bool {
-	var jitter time.Duration
-	if svc.config.Osquery.MaxJitterPercent > 0 {
-		maxJitter := int64(svc.config.Osquery.MaxJitterPercent) * int64(interval) / 100.0
-		jitter = time.Duration((int64(hostID) + svc.getJitterSeed()) % maxJitter)
+	svc.jitterMu.Lock()
+	defer svc.jitterMu.Unlock()
+
+	if svc.jitterH[interval] == nil {
+		svc.jitterH[interval] = &jitterHashTable{
+			maxCapacity: 1,
+			bucketCount: int(int64(svc.config.Osquery.MaxJitterPercent) * int64(interval.Minutes()) / 100.0),
+			buckets:     make(map[int]int),
+			cache:       make(map[uint]time.Duration),
+		}
+		if svc.jitterH[interval].bucketCount == 0 {
+			svc.jitterH[interval].bucketCount++
```

Comment:
Nit: `svc.jitterH[interval].bucketCount = 1`

Reply:
Btw, we should test things work as expected when there's no jitter configured.

Reply:
Great catch, added a test for this.